src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -776,14 +793,13 @@ NewTransactionResponseDto.ClientIdEnum convertClientId(
                 ).orElseThrow(() -> new InvalidRequestException("Null value as input origin"));
     }
 
-    private String extractBinFromPan(RequestAuthorizationRequestDto requestAuthorizationRequestDto) {
+    private Mono<Optional<String>> extractBinFromPan(RequestAuthorizationRequestDto requestAuthorizationRequestDto) {
```

Comment:
rename this method?

Reply:
the point is that `bin` is not retrieved only from the `pan`, so better rename method name

Reply:
Fixed with d232dd40676e53de7e9bbbf5080c3e2f99d49f69!